src/ln/channelmanager.rs

```diff
@@ -1551,9 +1553,11 @@ impl ChannelMessageHandler for ChannelManager {
 							(chan.remove(), funding_msg, monitor_update)
 						},
 						Err(e) => {
-							//TODO: Possibly remove the channel depending on e.action
-							return Err(e);
-						}
+							let failure = chan.get_mut().force_shutdown();
+							short_to_id.remove(&chan.get_mut().get_short_channel_id().unwrap());
+							self.finish_force_close_channel(failure);
+							return Err(HandleError{err: e.err, action: e.action});
```

Comment:
Here the errors are bad keys or shows strange behavior of peer so we close and disconnect. But pushing event here is pretty messy with channel and events locks so we give error upstream (ugly I admit still thinking on another way)

Reply:
We're gonna need to handle this in a number of places, so it may make sense to start thinking about splitting out the "Channel returned an Err, maybe call force_close and then return" function and wrap all the handle_MSG functions so that we can handle things appropriately.

Reply:
I was thinking more that the ChannelMessageHandler impl for ChannelManager would mostly consist of wrapper functions that called into "real" handler functions and then, on error, called force_close channel. The ChannelMessageHandler requirements say that we'll never have paralell message handler calls for the same peer, so locking should be fine in that we'll handle the error in order for the given peer.